test/matchers/matchers.go

```diff
@@ -0,0 +1,76 @@
+package matchers
+
+import (
+	"github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func HaveBackendTypeNats(tp v1alpha1.Backend) gomegatypes.GomegaMatcher {
+	return gomega.And(
+		gomega.WithTransform(
+			func(e *v1alpha1.Eventing) string {
+				return e.Spec.Backends.Type
+			}, gomega.Equal(tp.Type)),
```

Comment:
Because the function name says `HaveBackendTypeNats`, does it make sense to always check for the type "NATS" a as the required type here?

Reply:
If I always check for `NATS` here, I'd have to remove `Type` out of the `defaultBackend` function in the `integration_test.go` file and rename `defaultBackend` to `defaultBackendConfig`.

I prefer the implementation as is, because in the `defaultBackend` function now you can see in one glance how the complete default for `Backend` looks like.

What do you think?

Reply:
What happens if this func is used with a backend instance that has all the NATS required properties, but has the backend type set to EventMesh?

Reply:
What happens would depend on the testcase your using this matcher with. But I guess you mean, we should avoid the possibility to use this matcher wrongly?

Reply:
Done

Reply:
> we should avoid the possibility to use this matcher wrongly?

Yes, also because its name says `HaveBackendTypeNats` so we have to guarantee that NATS as a type is used as the desired type matcher.